test/libsolidity/SemanticTest.cpp

```diff
@@ -84,157 +84,164 @@ SemanticTest::SemanticTest(string const& _filename, langutil::EVMVersion _evmVer
+				m_compileViaYul = compileViaYul;
+				m_compileToEwasm = compileToEwasm;
+				m_compileViaYulCanBeSet = false;
 
-			if (compileViaYul)
-				AnsiColorized(_stream, _formatted, {BOLD, CYAN}) << _linePrefix << "Running via Yul:" << endl;
+				reset(m_compileToEwasm);
 
-			for (auto& test: m_tests)
-				test.reset();
+				if (compileViaYul)
+					AnsiColorized(_stream, _formatted, {BOLD, CYAN})
+						<< _linePrefix << "Running via Yul" << (m_compileToEwasm ? " (ewasm):" : ":") << endl;
 
-			map<string, solidity::test::Address> libraries;
+				for (auto& test: m_tests)
+					test.reset();
 
-			bool constructed = false;
+				map<string, solidity::test::Address> libraries;
 
-			for (auto& test: m_tests)
-			{
-				if (constructed)
-				{
-					soltestAssert(!test.call().isLibrary, "Libraries have to be deployed before any other call.");
-					soltestAssert(!test.call().isConstructor, "Constructor has to be the first function call expect for library deployments.");
-				}
-				else if (test.call().isLibrary)
+				bool constructed = false;
+
+				for (auto& test: m_tests)
 				{
-					soltestAssert(
+					if (constructed)
+					{
+						soltestAssert(!test.call().isLibrary, "Libraries have to be deployed before any other call.");
+						soltestAssert(
+							!test.call().isConstructor,
+							"Constructor has to be the first function call expect for library deployments.");
+					}
+					else if (test.call().isLibrary)
+					{
+						soltestAssert(
 							deploy(test.call().signature, 0, {}, libraries) && m_transactionSuccessful,
-							"Failed to deploy library " + test.call().signature
-					);
-					libraries[test.call().signature] = m_contractAddress;
-					continue;
-				}
-				else
-				{
+							"Failed to deploy library " + test.call().signature);
+						libraries[test.call().signature] = m_contractAddress;
+						continue;
+					}
+					else
+					{
+						if (test.call().isConstructor)
+							deploy("", test.call().value.value, test.call().arguments.rawBytes(), libraries);
+						else
+							soltestAssert(deploy("", 0, bytes(), libraries), "Failed to deploy contract.");
+						constructed = true;
+					}
+
 					if (test.call().isConstructor)
-						deploy("", test.call().value.value, test.call().arguments.rawBytes(), libraries);
+					{
+						if (m_transactionSuccessful == test.call().expectations.failure)
+							success = false;
+
+						test.setFailure(!m_transactionSuccessful);
+						test.setRawBytes(bytes());
+					}
 					else
-						soltestAssert(deploy("", 0, bytes(), libraries), "Failed to deploy contract.");
-					constructed = true;
+					{
+						bytes output;
+						if (test.call().useCallWithoutSignature)
+							output = callLowLevel(test.call().arguments.rawBytes(), test.call().value.value);
+						else
+						{
+							soltestAssert(
+								m_allowNonExistingFunctions
+									|| m_compiler.methodIdentifiers(m_compiler.lastContractName())
+										   .isMember(test.call().signature),
```

Comment:
This is quite some indentation depth. Can we somehow fix this by introducing helper functions?